File: src/Backend/InventarioEscolar.Domain/Entities/AssetMovement.cs
namespace InventarioEscolar.Domain.Entities
{
    public class AssetMovement : EntityTenantBase
    {
        public long AssetId { get; set; }
        public required Asset Asset { get; set; } 
        public long FromRoomId { get; set; }
        public required RoomLocation FromRoom { get; set; } 
        public long ToRoomId { get; set; }
        public required RoomLocation ToRoom { get; set; } 
        public DateTime MovedAt { get; set; } = DateTime.UtcNow;
        public string? Responsible { get; set; }
        public bool IsCanceled { get; set; } = false;
        public string? CancelReason { get; set; }
        public DateTime? CanceledAt { get; set; }
    }
}

File: frontend/EscolarInventarioApp/src/app/core/Models/AssetMovement.ts
import { Asset } from "./Asset";
import { BaseEntity } from "./BaseEntity";
import { RoomLocation } from "./RoomLocation";

export interface AssetMovement extends BaseEntity {
  asset?: Asset;
  fromRoom?: RoomLocation;
  toRoom?: RoomLocation;
  movedAt: Date;
  responsible?: string;
  isCanceled: boolean;
  canceledAt?: Date
  cancelReason?: string;
}
